### PR TITLE
Contra refactor

### PR DESCRIPTION
**Pull request by noroadsleft**
Readme gains a link.

Keyboard refactor:

- `LAYOUT_grid` is now `LAYOUT_ortho_4x12`
- `LAYOUT_mit` is now `LAYOUT_planck_mit`
- `LAYOUT_kc_grid` is now `LAYOUT_kc_ortho_4x12`
- `rules.mk` loses the `planck_grid` layou, which doesn't exist

Keymap refactor:

- Keymaps updated to use `#include QMK_KEYBOARD_H` and layout macros.

### CODE REVIEW

File: keyboards/contra/keymaps/ryanm101/keymap.c [hunk 1]
```diff
@@ -18,13 +17,13 @@ extern keymap_config_t keymap_config;
 #define _L2  3
 #define _L3  4
 
-enum custom_keycodes {
```
Comment (drashna) [inline keyboards/contra/keymaps/ryanm101/keymap.c]: Curly bracket got edited out. Needs to be there

File: keyboards/contra/keymaps/ryanm101/keymap.c [hunk 2]
```diff
@@ -18,13 +17,13 @@ extern keymap_config_t keymap_config;
...
+enum custom_keycodes
   QWERTY = SAFE_RANGE,
   QWERTYMAC,
   L1,
   L2,
   L3
-};
+;
```
Comment (drashna) [inline keyboards/contra/keymaps/ryanm101/keymap.c]: Curly bracket got edited out. Needs to be there

File: keyboards/contra/keymaps/ryanm101/keymap.c [hunk 3]
```diff
@@ -37,7 +36,7 @@ enum custom_keycodes {
 #define CADCAE TD(CAD_CAE)
 #define SHIFTSLASHPIPE TD(TD_SHIFTSLASHPIPE)
 
-enum {
```
Comment (drashna) [inline keyboards/contra/keymaps/ryanm101/keymap.c]: Curly bracket got edited out. Needs to be there

File: keyboards/contra/keymaps/ryanm101/keymap.c [hunk 4]
```diff
@@ -47,208 +46,208 @@ enum {
...
+    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, KC_SPC,  XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX
+  )
+
+;
 
 // Parantheses
-void paranthesis_dance (qk_tap_dance_state_t *state, void *user_data) {
-  if (state->count == 1) {
+void paranthesis_dance (qk_tap_dance_state_t *state, void *user_data)
+  if (state->count == 1)
```
Comment (drashna) [inline keyboards/contra/keymaps/ryanm101/keymap.c]: Curly bracket got edited out. Needs to be there

File: keyboards/contra/keymaps/ryanm101/keymap.c [hunk 5]
```diff
@@ -47,208 +46,208 @@ enum {
...
     SEND_STRING("()"); register_code(KC_LEFT); unregister_code(KC_LEFT);
-  } else if (state->count == 2) {
```
Comment (drashna) [inline keyboards/contra/keymaps/ryanm101/keymap.c]: Curly bracket got edited out. Needs to be there

File: keyboards/contra/keymaps/ryanm101/keymap.c [hunk 6]
```diff
@@ -47,208 +46,208 @@ enum {
...
+   else if (state->count == 2)
     SEND_STRING("(");
-  } else if (state->count == 3) {
```
Comment (drashna) [inline keyboards/contra/keymaps/ryanm101/keymap.c]: Curly bracket got edited out. Needs to be there

**Comment by drashna** [comment]
Also, it may be useful to use the `LAYOUTS_HAS_RGB` method that we use for bakingpy's keymap. 
https://github.com/qmk/qmk_firmware/blob/master/layouts/community/ortho_4x12/bakingpy/rules.mk

File: keyboards/contra/keymaps/ryanm101/keymap.c (hunk 6 above)
Comment (noroadsleft) [reply keyboards/contra/keymaps/ryanm101/keymap.c]: Definitely was trying to remove in a selection and had the option disabled.

**Comment by noroadsleft** [comment]
@drashna, re: `LAYOUTS_HAS_RGB`:

1. Is it needed? And if so:
2. How do I implement it?

Have no experience with this method outside of knowing it exists.

**Comment by noroadsleft** [comment]
@drashna `contra.h` review isn't clearing because you tagged Line 30 when the issue was actually on Line 36. But it's fixed.

**Comment by drashna** [comment]
The `LAYOUT_HAS_RGB` is this: https://github.com/qmk/qmk_firmware/blob/master/layouts/community/ortho_4x12/bakingpy/rules.mk

So, replace this line with the above: https://github.com/qmk/qmk_firmware/blob/master/layouts/community/ortho_4x12/xyverz/rules.mk#L2

File: keyboards/contra/keymaps/ryanm101/keymap.c [hunk 7]
```diff
@@ -50,97 +50,98 @@ enum {
...
+   * `-----------------------------------------------------------------------------------'
+   */
+  [_L3] = LAYOUT_ortho_4x12(
+    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, KC_PGDN, KC_PGUP, XXXXXXX, XXXXXXX, XXXXXXX,
+    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT, XXXXXXX, XXXXXXX,
+    TMUX,    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, KC_ENT,
+    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, KC_SPC,  XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX
+  )
+
+);
```
Comment (drashna) [inline keyboards/contra/keymaps/ryanm101/keymap.c]: This should be a curly bracket

Comment (noroadsleft) [reply keyboards/contra/keymaps/ryanm101/keymap.c]: :roll_eyes: 

:frowning_face: :gun:

**Comment by drashna** [comment]
Looks like the only issues remaining are not related to this board. (Layout and userspace issues)

**Comment by drashna** [comment]
Thanks!

**Pull request merged**